src/main/java/macrobase/analysis/outlier/OutlierClassifier.java

```diff
@@ -0,0 +1,19 @@
+package macrobase.analysis.outlier;
+
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+
+public abstract class OutlierClassifier {
+
+    private OutlierClassificationResult lastResult;
+
+    public OutlierClassifier(MacroBaseConf conf) {}
+    public abstract void observe(Datum datumToClassify);
+
+    public abstract OutlierClassificationResult classify();
```

Comment:
I think that if OutlierClassificationResult has both `Datum getDatum()` and `boolean isOutlier()` functions, this will be clearer.